### PR TITLE
Add tests for primtives usage with algorithms and aer

### PR DESCRIPTION
**Pull request by mtreinish**
This commit adds some new tests that exercise the primtives API with algorithms is used. In Qiskit Terra 0.22.0 breaking API changes merged to the primitives API, while it is too late to roll them back we can prevent such user impacting breaking changes in the future by ensuring we have integration test coverage in qiskit-neko. Ths commit adds tests that use the BackendSampler and BackendEstimator primitives with SamplingVQE and VQE to test the backend interface, and then also to ensure we're validating the cross-project compatibility with the abstract primitive interface tests that use the specific Aer implementation of the primitives API is added. This will ensure that no change to qiskit-terra can merge which will break an external implementation of the primitives API in the future.

A follow on to this is to add support for configurable primitives similar to what we already have with backends. This will enable adding test jobs that use a provider's native primitive implenetations for primitive tests.

### CODE REVIEW

**Comment by mtreinish** [comment]
This will be blocked until qiskit-aer 0.11.1 is released because the primitives in aer are incompatible with qiskit-terra 0.22.0 because of the breaking API changes. We should strive to merge this soon after it's released though to prevent this kind of thing in the future.

File: qiskit_neko/tests/primitives/test_vqe.py
Comment (jakelishman) [inline qiskit_neko/tests/primitives/test_vqe.py:38]: the docstring and the code don't agree here!

**Review by jakelishman — APPROVED** [review]
I'm fully in favour of getting something that exercises the primitives API in the "blocking" pipeline for Terra CI in a way that forces adherence to the deprecation policy.

File: qiskit_neko/tests/primitives/test_vqe.py
Comment (mtreinish) [reply qiskit_neko/tests/primitives/test_vqe.py:38]: lol, I'm too lazy with my copy and paste

```suggestion
        """Test the execution of SamplingVQE with BackendSampler."""
        sampler = BackendSampler(self.backend)
```

**Review by jakelishman — APPROVED** [review]

**Comment by mtreinish** [comment]
Sigh, I broke the tests with https://github.com/Qiskit/qiskit-terra/pull/8955, apparently BackendV1 backends didn't have to specify `max_experiments` (and aer doesn't). So that's causing the test with the configured backend to fail (because by default it's just `AerSimulator`).

**Pull request merged**
